tsp-output/@azure-tools/typespec-csharp/azoai/AzureOpenAI/Models/Input/CreateChatCompletionRequestExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿using OpenAI.Models;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Diagnostics;
+
+namespace AzureOpenAI.Models;
+
+public static class CreateChatCompletionRequestExtensions
+{
+    public static IList<AzureChatExtensionConfiguration> GetDataSources(this CreateChatCompletionRequest request)
+    {
+        if (request is not IJsonModel model)
+        {
+            throw new InvalidOperationException("TODO");
+        }
+
+        JsonModelList<AzureChatExtensionConfiguration> dataSources;
+
+        if (model.AdditionalProperties.TryGetValue("data_sources", out object? value))
```

Comment:
Retrieves value from `AdditionalProperties` dictionary, or creates it new and adds it to the dictionary.